src/stdlib_ascii.f90

```diff
@@ -277,23 +277,30 @@ end function to_upper
     pure function to_title(string) result(title_string)
         character(len=*), intent(in) :: string
         character(len=len(string)) :: title_string
+        logical:: capitalize_flag
         integer :: i, n
 
         n = len(string)
+        capitalize_flag = .TRUE.
         do i = 1, len(string)
-            if (is_alphanum(string(i:i))) then
-                title_string(i:i) = char_to_upper(string(i:i))
-                n = i
-                exit
+
+            if(capitalize_flag) then
+                if (is_alphanum(string(i:i))) then
+                    title_string(i:i) = char_to_upper(string(i:i))
+                    capitalize_flag = .FALSE.
+                else
+                    title_string(i:i) = char_to_lower(string(i:i))
+                end if
             else
-                title_string(i:i) = string(i:i)
+                if(string(i:i)==" ") then
```

Comment:
This could be more general than just spaces, every non-word boundary might trigger capitalization, consider:
`to_title(".this.sentence.is.separated.with.dots.")`, would we expect `.This.sentence.is.separated.with.dots.` or  `.This.Sentence.Is.Separated.With.Dots.` for a titlecase return value?

Reply:
Hmmm! I am not sure what to expect in such a case. My favor would go to capitalize each word. What do other languages?

Reply:
I checked for python and realised that every character which is not an alphabet triggers capitalization including integers. Python mentions that even apostrophe(`'`) triggers this capitalization which is undesirable. May be we can take an optional set as the argument from the user which will consist of characters the user wants to the trigger capitalization for..